### ISSUE TITLE
Admonition bleeding out with extra pixels in Firefox

### ISSUE THREAD

**Issue by jmaneyrol69** [squidfunk/mkdocs-material#3782]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

There are some extra pixels on the border of admonition boxes on Firefox (98.0.2) Linux (Ubuntu 20.04LTS):

![Firefox-mkdocs-boxes](https://user-images.githubusercontent.com/45173285/160788364-29c6ba08-3d79-420c-9fff-966fca62cad7.png)

This does not happen with Google Chrome nor Chromium Linux:

![Chrome-mkdocs-boxes](https://user-images.githubusercontent.com/45173285/160788471-f7a39ddc-b418-42cf-ab2d-967c8643318b.png)

Also, this seems to only affects Firefox on Linux, it does not occur on Windows (other OSes not tested).

### Expected behaviour

There shouldn't be any extra pixels:
![Chrome-mkdocs-boxes](https://user-images.githubusercontent.com/45173285/160788729-5dcd5540-a0f8-4e72-989e-cff4395f9554.png)

### Actual behaviour

There are extra pixels in the border of the title area:
![Firefox-mkdocs-boxes](https://user-images.githubusercontent.com/45173285/160788831-d181be7f-7d7b-444e-a843-932b40fc4fab.png)

### Steps to reproduce

Use Firefox 98.0.2 on Ubuntu 20.04LTS and go to https://squidfunk.github.io/mkdocs-material/reference/admonitions/?h=admoni#supported-types

### Package versions

- Python: 3.8.10
- MkDocs: 1.2.3
- Material: 8.2.5

### Configuration

```yaml
site_name: Material for MkDocs
site_url: https://squidfunk.github.io/mkdocs-material/
```

### System information

- Operating system: Ubuntu 20.04LTS
- Browser: Firefox 98.0.2

**Comment by squidfunk** [comment]
Thanks for reporting. I can reproduce this behavior on the latest macOS Firefox as well if the viewport is wider than 1600px. It seems to be related to rounding errors and applies to admonitions as well as details. If you remove the border, on the left, the titles are still wider than the container, which can be seen if we give the background another color:

<img width="475" alt="Bildschirmfoto 2022-03-30 um 11 40 40" src="https://user-images.githubusercontent.com/932156/160802609-192d5d30-afdd-4084-bef3-a281ddfb28a9.png">

I'm unsure if this is fixable. I'll look into it when I find some time.

**Comment by iBug** [comment]
Just to add that Firefox on Windows 10 is also affected. Similar to macOS, a wide screen is required for this to trigger.

**Comment by squidfunk** [comment]
@iBug Thanks. Could you please check if the fix resolves the issue?

**Comment by iBug** [comment]
I'm afraid it's going wrong on the other side.

I'm using Firefox 97.0.1 on Windows 10 21H2, with monitor size 2560×1440 and system zoom at 100%. My dev tools open to the right (as the default on Chrome and Edge).

## Without patches

Visit the [Getting Started](https://squidfunk.github.io/mkdocs-material/getting-started/) page at *some* browser zoom levels trigger the problem, specifically:

- Any viewport width looks fine on 100% zoom.
- With `100vw = 1800px`, page zoom 40% 60%, 110%, 120%, 220% trigger the "left bleed-out" effect:

	![image](https://user-images.githubusercontent.com/7273074/161436021-6424dc59-d966-486e-8837-65e8b77af54e.png)

	Note that 120% page zoom seems to be an easily reproducible setup.

- With certain setup, e.g. 120% zoom at `100vw = 1970px` (Firefox shows 1642px after zooming in), there's a "right bleed-out effect":

	![image](https://user-images.githubusercontent.com/7273074/161436114-d2e78b4f-308d-4952-9db0-6aebf6b1e8d9.png)

	Again this is somewhat easy to trigger on 120% zoom. Just drag your dev tools to change viewport width.

## With patch

I'm not an MkDocs theme developer so I don't know the "canonical" way to apply c5a06c1d82c8f1c0d570e55c8ae8313a4ed69e1a. Here's what I did instead:

- Read the code: It effectively adds one CSS rule for Firefox:

	```css
	margin-inline-start: px2rem(-15.5px);
	```

- Find out the value for `1rem`. In my case it's 20px.
- Do the calculation of `px2rem` manually. I got `-.775rem`.
- Locate any `.md-typeset :is(.admonition-title, summary)` entry in dev tools, insert this rule:

	```css
	margin-inline-start: -.775rem;
	```

	Make sure the new rule is not <s>slashed out</s>.

### Result

I could not observe any differences, after repeated ticking and unticking the newly added rule:

![image](https://user-images.githubusercontent.com/7273074/161436438-d33c1066-72b9-429a-a75e-1e3fb7aa0fbc.png)

**Comment by iBug** [comment]
Pretty sure the reason it doesn't work is that the new patch changed effectively nothing. This is already `-0.775rem` (not manually added):

![image](https://user-images.githubusercontent.com/7273074/161436555-97c33bcd-e228-4005-bd80-025fa5fdd963.png)

**Comment by squidfunk** [comment]
Okay. Thanks for testing. My testing showed that everything worked, but I guess it is may also be impacted by the resolution of the monitor, as well as display width. I'm not sure this can be fixed then, because the rounding errors will always manifest to some degree in certain situations. As a mitigation, we could remove the border so that the issue is not as perceptible as it currently is, but the issue at hand won't fix.

**Comment by squidfunk** [comment]
@iBug only to be sure, could you please check the admonition reference? It has the patch applied – only to rule out that you missed a step when testing the patch manually:

https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage

**Comment by iBug** [comment]
@squidfunk I'm afraid no, again. Both screenshots are taken at 100% system zoom and 110% page zoom (browser zoom) on my 2560×1440 monitor at different viewport widths.

> ![image](https://user-images.githubusercontent.com/7273074/161490444-742c7b5a-11d3-4625-9a8c-73125e4ba448.png) ![image](https://user-images.githubusercontent.com/7273074/161490503-8aa9f026-13e0-443e-9428-056593178810.png)

My Firefox auto-updated to 98.0.2 after yesterday's testing but I believe it makes no difference.

**Comment by iBug** [comment]
HOWEVER I found the problem gone after removing (unchecking the checkbox in F12 Dev Tools) *both* instances of this rule:

```css
.md-typeset :is(.admonition-title, summary) {
  border: 0 solid #448aff;	
}
```

Except that the admonition title goes a bit to the left:

> ![image](https://user-images.githubusercontent.com/7273074/161491205-31fb2675-5999-42d1-ae4f-e98c41d8e0cc.png) ![image](https://user-images.githubusercontent.com/7273074/161491237-ff88ee82-09b5-49a4-be72-f66e67ba8a8e.png)

**Comment by squidfunk** [comment]
@iBug the problem is not gone. If you look closely, the background will now bleed out, but it is barely noticeable.

**Comment by squidfunk** [comment]
Fixed in 0f7e9fad0. I've removed the border and adjusted the positioning so that it looks identical to the current solution. @facelessuser IIRC, this is likely going to break your customizations, as it basically reverts https://github.com/squidfunk/mkdocs-material/commit/8eb2375551fc1542d6969678d0b31614152a4bb3, after which you had to adjust your styles. The reason why the title is moved all up to the left and does not stop at the border is because if we start at the border, the focus outline looks weird on `details` elements. Since I do not want to maintain different styles for admonitions and `details`, I normalized it. With the current change, the outline is preserved.
